pkgs/by-name/bo/boring/package.nix

```diff
@@ -0,0 +1,44 @@
+{
+  buildGoModule,
+  fetchFromGitHub,
+  lib,
+}:
+
+buildGoModule rec {
+  pname = "boring";
+  version = "0.11.4";
+
+  src = fetchFromGitHub {
+    owner = "alebeck";
+    repo = "boring";
+    rev = version;
+    hash = "sha256-A9ps85vQQPibLuYMmVcPdRzA25+V1XZnX2qiqGTaPyc=";
+    leaveDotGit = true;
+    postFetch = ''
+      git -C "$out" rev-parse --short HEAD 2> /dev/null > "$out/commit"
+      rm -rf "$out/.git"
+    '';
+  };
+
+  vendorHash = "sha256-j8A0F+o3EnzJdge+T/gHAwRGwzC86oD6ddZejUs/C7o=";
+
+  ldflags = [
+    "-s"
+    "-w"
+    "-X main.version=${version}"
+  ];
+
+  preBuild = ''
+    ldflags+=" -X main.commit=$(cat commit)"
```

Comment:
question:

Using finalAttrs.src.rev seems like an easy way to get the version. However, is this trick really used for version numbers?

```console
> ./results/boring-x86_64-linux/bin/boring version
boring 0.11.4
```

https://github.com/alebeck/boring/blob/b98c1b492cb2ad1cf90449eaaffdbce0d1054a89/cmd/boring/main.go#L84-L93

Is the commit revision only applied if no version number is present? If so, we can simply remove this ldflag.